test-app/tests/integration/components/checkbox-test.gts

```diff
@@ -95,6 +107,43 @@ module('Integration | Component | Checkbox', function (hooks) {
     assert.dom('[data-checkbox]').hasNoClass('bg-primary-idle');
   });
 
+  test('it applies the expected classes when `@isChecked={{true}}` and `@isReadOnly={{true}}', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <CheckboxControl @isChecked={{true}} @isReadOnly={{true}} data-checkbox />
+    </template>);
+
+    assert.dom('[data-checkbox]').hasClass('bg-titles-and-attributes');
+    assert.dom('[data-checkbox]').hasClass('border-none');
+    assert.dom('[data-checkbox]').hasNoClass('bg-primary-idle');
+    assert.dom('[data-checkbox]').hasNoClass('bg-transparent');
+    assert.dom('[data-checkbox]').hasNoClass('bg-surface-xl');
+    assert.dom('[data-checkbox]').hasNoClass('bg-normal-idle');
+    assert.dom('[data-checkbox]').hasNoClass('bg-disabled');
+    assert.dom('[data-checkbox]').hasNoClass('border-disabled');
```

Comment:
Visual regression testing when 😅 

This is okay for now so we can ensure we are applying the correct classes, but we wouldn't need these assertions at all with something like Percy.  Maybe soon™.

Reply:
We are not blocked on introducing Percy on OSS Github repos, are we? They have a free plan, which might be enough, but I think there is also a special program for OSS: https://www.browserstack.com/open-source

Reply:
I don't think so, no! 🎉 We may need to seek approval from someone to add it but I'd imagine it'd be pretty quick. Let me see if I can get that process started today.